### PR TITLE
[MSHARED-893] deprecate more methods and clean up api doc and tests

### PR DESCRIPTION
**Pull request by elharo**
@michael-o I think this closes out all the FileUtils methods that can be straight-forwardly replaced in Java 7+.

### CODE REVIEW

File: src/main/java/org/apache/maven/shared/utils/io/FileUtils.java [hunk 1]
```diff
@@ -2103,10 +2117,10 @@ public static boolean isSymbolicLinkForSure( @Nonnull final File file )
...
-     * @param symlink The link name.
-     * @param target The target.
-     * @return The linked file.
-     * @throws IOException in case of an error.
+     * @param symlink the link name
+     * @param target the target
+     * @return the linked file
+     * @throws IOException in case of an error
      */
     @Nonnull public static File createSymbolicLink( @Nonnull File symlink,  @Nonnull File target )
```
Comment (michael-o) [inline src/main/java/org/apache/maven/shared/utils/io/FileUtils.java:2125]: What about `java.nio.file.Files.createSymbolicLink(Path, Path, FileAttribute<?>...)`?

Comment (elharo) [reply src/main/java/org/apache/maven/shared/utils/io/FileUtils.java:2125]: Perhaps. This method is slightly different, particularly in what happens when the symbolic link already exists. It wraps the `java.nio.file.Files.createSymbolicLink` method for reasons that aren't completely clear to me.

Comment (michael-o) [reply src/main/java/org/apache/maven/shared/utils/io/FileUtils.java:2125]: OK, I would at least add a `@see` or deprecate and indicate the different behavior.

Comment (elharo) [reply src/main/java/org/apache/maven/shared/utils/io/FileUtils.java:2129]: done

**Pull request merged**